src/webapp/components/table/BasicTable.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { Table, TableBody, TableCell, TableHead, TableRow, Link } from "@material-ui/core";
+import styled from "styled-components";
+import { Maybe } from "../../../utils/ts-utils";
+import { Selector } from "../selector/Selector";
+import i18n from "../../../utils/i18n";
+import { SelectorOption } from "../selector/utils/selectorHelper";
+
+interface BaseColumn {
+    value: string;
+    label: string;
+    underline?: boolean;
+}
+interface LinkColumn extends BaseColumn {
+    type: "link";
+}
+interface SelectorColumn extends BaseColumn {
+    type: "selector";
+    options: SelectorOption[];
+}
+
+export type TableColumn = BaseColumn | LinkColumn | SelectorColumn;
+interface BasicTableProps {
+    columns: TableColumn[];
+    rows: {
+        [key: TableColumn["value"]]: string;
+    }[];
+    onChange?: (cell: Maybe<string>, rowIndex: number, column: TableColumn["value"]) => void;
+    showRowIndex?: boolean;
+}
+
+export const BasicTable: React.FC<BasicTableProps> = React.memo(
+    ({ columns, rows, onChange = () => {}, showRowIndex = false }) => {
+        const Cell = (cell: string, rowIndex: number, column: TableColumn) => {
+            const [selectorValue, setSelectorValue] = React.useState<string>(cell);
+            if ("type" in column && column.type === "link") {
+                return (
+                    <StyledLink onClick={() => onChange(cell, rowIndex, column.value)}>
+                        {cell}
+                    </StyledLink>
+                );
+            } else if ("type" in column && column.type === "selector") {
+                const handleChange = (value: string) => {
```

Comment:
Also if you change this to a new component remember to use useCallback (or useMemo) to avoid unnecesary re-renders

Reply:
I wrap Cell component with `React.memo`